dipy/io/stateful_tractogram.py

```diff
@@ -96,11 +103,25 @@ def __init__(self, streamlines, reference, space,
                                       data_per_point=data_per_point,
                                       data_per_streamline=data_per_streamline)
 
-        space_attributes = get_reference_info(reference)
-        if space_attributes is None:
-            raise TypeError('Reference MUST be one of the following:\n' +
-                            'Nifti or Trk filename, Nifti1Image or TrkFile, ' +
-                            'Nifti1Header or trk.header (dict)')
+        if isinstance(reference, dipy.io.stateful_tractogram.StatefulTractogram):
+            logging.warning('Using a StatefulTractogram as reference, this '
+                            'will copy only the space_attributes, not '
+                            'the state. The variables space and origin_at_corner '
+                            'must be specified separately.')
+
+        if isinstance(reference, tuple) and len(reference) == 4:
+            if is_reference_info_valid(*reference):
+                space_attributes = reference
+            else:
+                raise TypeError('The provided space attributes are not '
```

Comment:
I guess you can be a bit more specific here? If `is_reference_info_valid` were to return something more about why it's returning false?

Reply:
The function does provide all the `logging.warning`, which would likely be displayed (unless the user turn off logging entirely). An user would see every failed verification before the crash as logging.